x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/watch/WatchStoreUtils.java

```diff
@@ -28,11 +29,18 @@ public static IndexMetadata getConcreteIndex(String name, Metadata metadata) {
             return null;
         }
 
-        if (indexAbstraction.getType() != IndexAbstraction.Type.CONCRETE_INDEX && indexAbstraction.getIndices().size() > 1) {
+        if ((indexAbstraction.getType() != IndexAbstraction.Type.CONCRETE_INDEX
```

Comment:
This if statement does look a bit complicated.
Can we instead have two simpler if statements for the valid cases?
One if statement for when `indexAbstraction` is a data stream and one when `indexAbstraction` is an alias.
In all other cases we would fail.

Reply:
I think we also support if it's just a concrete index, right? I was not sure if there were other types I didn't know about so I left it mostly as it was. If there are definitely just those 3 though (alias, data stream, concrete index) I can definitely simplify it.

Reply:
In at least one place we pass `.watches` to this method, which I believe is a concrete index (although I could definitely be wrong about that).

Reply:
Thanks, it does.